python/tests/test_cats.py

```diff
@@ -17,3 +17,23 @@ def test_cats():
     assert action >= 10
     assert action <= 20
     vw.finish()
+
+def test_cats_pdf():
+    min_value = 10
+    max_value = 20
+
+    vw = pyvw.vw("--cats_pdf 4 --min_value " + str(min_value) + " --max_value " + str(max_value) + " --bandwidth 1")
+    vw_example = vw.parse("ca 15:0.657567:6.20426e-05 | f1 f2 f3 f4", pyvw.vw.lContinuous)
+    vw.learn(vw_example)
+    vw.finish_example(vw_example)
+
+    assert vw.get_prediction_type() == vw.pPDF, "prediction_type should be pdf"
+
+    pdf_segments = vw.predict("| f1 f2 f3 f4")
+    for segment in pdf_segments:
+        assert len(segment) == 3
```

Comment:
What is the 3rd segment? Should it be asserted too?

Reply:
The 3rd value in a segment is the [pdf_value](https://github.com/VowpalWabbit/vowpal_wabbit/wiki/CATS,-CATS-pdf-for-Continuous-Actions#prediction-type) and the only constraint on it is that is should be non-negative. 

I wasn't sure what level of assertions I should add. For instance, I could have asserted the exact segment values themselves. I chose to follow what the existing `test_cats` function in the same file did -- only test if the action is in the expected range.

Reply:
One test we could add here would be to check that the returned pdf is valid, see [here](https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/vowpalwabbit/prob_dist_cont.cc#L51)

Reply:
Makes sense. Added an assertion for checking pdf mass ~= 1. (while at it, added one for pdf value being non-negative)